### DISCUSSION TITLE
Zoom livestream on Debian with high quality audio using desktop audio sharing

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#1209] [General]
Probably that's my first own post here ;-).

I think the Jamulus community might know how to do this:

We (= my choir and I) are planning to do a Zoom video conference with a live Jamulus rehearsal. It should be possible for the Zoom attendants to talk to the Jamulus singers and of course the other way round. Furthermore, I'd like to be able to turn the Jamulus users down for Zoom, and the other way round while still being able to hear/talk to the other group. Basically I'm in the audio engineering position.

To allow this, I've set up JACK routing via Ardour to Zoom by using a PulseAudio <--> JACK bridge. I've tested my setup with JACK and Ardour today and "it seems to work":
![RoutingJack](https://user-images.githubusercontent.com/20726856/110381139-5ca9ab00-8059-11eb-9a07-ef8d928b9baf.png)
I hope this complex routing does what I intend it to do...

The only problem I have at the moment is audio quality in Zoom. Since it seems to be impossible to set the audio quality to HiFi mode on Linux (and I don't want to use Windows or macOS for audio work), audio gets sent to Zoom through the normal microphone inputs which seem to apply all the audio "enhancements" you wouldn't want to have for music. I know that it is possible to share stereo desktop audio via Zoom on Debian but I haven't found a way to get the audio from JACK to the input for desktop audio in Zoom. Does anybody more experienced than me know how to hack this input? Does the PulseAudio bridge support something like that?

Edit: I just found this: https://www.reddit.com/r/linuxaudio/comments/lzx8l5/route_audio_from_jack_to_zoom_without_pre/

**Comment by hoffie** [comment]
Not using Zoom, but my experience from a similar setup with other video conferencing:
- The pulseaudio jack plugin works.
- The routing was rather simple for me (Jamulus <-> Firefox via Pulseaudio-Jack, I think it's more complicated in your case as you seem to be running Ardour in addition?
- The main things which broke audio quality for me were "speech enhancement" techniques such as noise suppresion and automatic gain control. Not sure if you are talking about a native Zoom client or about using Zoom via Firefox (etc.)? I did the latter and disabled all audio enhancements there, which boosted quality a lot. These are the following settings in Firefox (about:config):

```
media.getusermedia.aec_enabled false
media.getusermedia.aec_extended_filter false
media.getusermedia.agc_enabled false
media.getusermedia.hpf_enabled false
media.getusermedia.noise_enabled false
```

**Reply by ann0see** [reply]
Thank you for this proposal. I was using GoToMeeting with Firefox some weeks ago and it made the PC almost non-responsive. Hopefully Zoom Web works better.

I use Ardour in between Jamulus/Zoom to make muting either Jamulus for zoom or zoom for Jamulus possible while still being able to hear the other tool via my headphones. Maybe there's an easier way to do this? I don't want to connect/disconnect in/outputs in Qjackctl or similar.

Background: A live concert over Zoom where Jamulus, Zoom and I can talk and hear the other side (but not always).

**Reply by reinhardwh** [reply]
Hi
We are using Jitsi
Freeware Videocall

Jamulus for Sound
Muted Jitsi


https://jitsi.org/what-is-jitsi/

**Reply by hoffie** [reply]
> I use Ardour in between Jamulus/Zoom to make muting either Jamulus for zoom or zoom for Jamulus possible while still being able to hear the other tool via my headphones. Maybe there's an easier way to do this?

I basically use the native features:
- Regular "Mute microphone" button in the video conference in order to mute the Jamulus-to-conference direction
- Regular "Mute myself" checkbox in Jamulus to mute the conference-to-Jamulus direction (alternatively: Firefox allows muting a Tab's audio output by clicking the little speaker in the Tab header)

I think this could still work with your headphones, although I haven't tried it that way.

My experiences cover Jitsi and WebEx. This bridge is running on a one-core virtual machine which I control via virt-manager and I had no performance issues. One should avoid getting any video over virt-manager as that kills (network) performance. :)

**Reply by ann0see** [reply]
I've now tried the zoom web client and sound quality is still bad. Jitsi with the correct options works a lot better (and much easier for me). Unfortunately I don't think we can give a jitsi link to 50+ people? I'll probably stream the concert with Jitsi and zoom web. People who can use Jitsi should use it and others can stay in zoom.

**Reply by ann0see** [reply]
Hmm. Trying to install zoom in a windows 8 qemu VM which I'm going to connect to JACK. Let's wait and see if this works. Unfortunately I need to compile qemu from source to get JACK support.

**Reply by ignotus666** [reply]
I'm not really sure I understand what you're trying to achieve but these past few days I also had a situation where I had to use Zoom in a Windows VM because it lacked a necessary functionality in the Linux version (conference interpreting). Also, I wanted the mic and headphone signals to go through Jack Mixer so I could control levels via a MIDI controller. The way I did it was to do all the audio signal routing in JACK in the host OS (Linux Mint) and the VM with Zoom simply picked up the signals via the pulseaudio bridge - no need to install JACK in the VM.

**Reply by ann0see** [reply]
Yes, I'll probably just use the pulseaudio bridge. Qemu seems to support JACK natively, but I just tried to compile qemu and didn't get the JACK support to work. Probably it's not worth the hassle.

**Comment by achim-grosse-oetringhaus** [comment]
Hello community,
I have set up a Jamulus-to-Zoom or Jamulus-to-Jitsi bridge for choir rehearsals where some members can use Jamulus and some can't. It uses a single computer, no audio hardware and no sophisticated software like DAW, only some pulseaudio commands and a few mouse clicks in QJackCtl.
Are you interested in me writing a step-by-step guide for the Tips & Tricks section (or for whereever)?
Best regards,
Achim

**Reply by jujudusud** [reply]
I am really interested in Jamulus to Jitsi.

**Reply by ann0see** [reply]
Yes! That would be great! I'd suggest to put it into the community knowledge base: https://jamulus.io/kb/index.html

You should use this link to add content: https://github.com/jamulussoftware/jamuluswebsite/new/release/_posts/?value=---%0Alayout%3A%20post%0Atitle%3A%20%22Your%20Title%22%0Alang%3A%20%22en%22%0Aauthor%3A%20%22YourName%22%0Aheading%3A%20%22Heading%22%0A---%0AName%20this%20file%20and%20edit%20the%20parameters%20above%21&message=New%20post

**Reply by WolfganP** [reply]
Good initiative, thx @achim-grosse-oetringhaus . Having some documentation would be very valuable for those who want to walk the same path.

**Reply by ann0see** [reply]
Windows would be great too. Unfortunately I don't have much experience there
